mail/mailer.go

```diff
@@ -243,6 +249,19 @@ func (di *dialerImpl) Dial() (smtpClient, error) {
 	return client, nil
 }
 
+// resetAndError resets the connection and then returns its argument as an error.
```

Comment:
I think s/resets the connection/resets the current mail transaction/ might be a bit clearer about what is actually happening here.